src/common/hashes/sha512hash.ts

```diff
@@ -0,0 +1,4 @@
+import {createHash} from 'crypto'
+export default (hex: string): string => {
```

Comment:
I personally find this syntax hard to parse at a glance. If you name the function rather than exporting something anonymous the code becomes more clear. 

Consider:
```
function sha512('hex string): string {
...
}

export default sha512;
```